packages/hopp-plugin-concat/lib/index.js

```diff
@@ -18,7 +18,35 @@ export const config = {
 
 /**
  * We don't need to do any real transformation.
+ * 
+ * suffix and prefix features
+ * @param suffix || null @param prefix || null
+ * appending additional suffix and prefix to each data.
  */
 export default async (ctx, data) => {
+  /**
+   * unexpected num of arguments handling
+   */
+  if (ctx.args.length > 1) {
+    throw new Error('Unexpected numbers of arguments.')
+  }
+
+  /**
+   * options assign
+   */
+  var opts = Object.assign({}, ctx.args[0] || {})
+
+  // suffix 
+  if (opts.suffix) {
+    data.size += opts.suffix.length
+    data.body = Buffer.from(opts.suffix + data.body.toString())
+  }
+
+  // prefix
+  if (opts.prefix) {
```

Comment:
Suffix should only be added if `data.end` is true (i.e. it is the last piece of data)